externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -191,4 +191,100 @@ def Tcp_BroadcastOp : Tcp_Op<"broadcast", [
   let hasVerifier = 1;
 }
 
+def Tcp_YieldOp : Tcp_Op<"yield", [Terminator, Pure]> {
+  let summary = "Yields values from regions in Tcp";
+
+  let description = [{
+    Terminates and yields values from regions in Tcp.
+
+    The values that are yielded should correspond to the parent operation's
+    results.
+  }];
+
+  let arguments = (ins
+    Variadic<Tcp_TensorOrScalar>:$ins
+  );
+
+  let builders = [
+    OpBuilder<(ins), [{ }]>,
+  ];
+
+  let assemblyFormat = "$ins attr-dict `:` type($ins)";
+}
+
+def Tcp_GroupOp : Tcp_Op<"group", [
+          AffineScope,
+          AutomaticAllocationScope,
+          ParentOneOf<["func::FuncOp", "tcp::GroupOp", "tcp::IsolatedGroupOp"]>,
+          SingleBlockImplicitTerminator<"tcp::YieldOp">]> {
+  let summary = "Groups ops into a region that is not isolated from above";
+
+  let description = [{
+    Provides a way to group ops into a region that is not isolated from above.
+
+    Since this is not isolated from above, ops inside this region can refer to
+    values in the outer scopes. So, this op does not have any input parameters.
+
+    The group_type attribute is a free-form string that can be used to denote
+    any aspect of the group. For example, it could be set to "elementwise" for
+    a group that only contains elementwise ops.
+
+    The enclosed region will terminate with a `tcp.yield` op. This ops results
+    should correspond to the values that are yielded by the terminating op.
+  }];
+
+  let arguments = (ins
+    StrAttr:$group_type
+  );
+
+  let results = (outs
+    Variadic<Tcp_TensorOrScalar>:$outs
+  );
+
+  let regions = (region
+    SizedRegion<1>:$body
+  );
+
+  let assemblyFormat = "attr-dict-with-keyword $body `:` type($outs)";
+}
+
+def Tcp_IsolatedGroupOp : Tcp_Op<"isolated_group", [
+          AffineScope,
+          AutomaticAllocationScope,
+          IsolatedFromAbove,
+          ParentOneOf<["func::FuncOp", "tcp::GroupOp", "tcp::IsolatedGroupOp"]>,
+          SingleBlockImplicitTerminator<"tcp::YieldOp">]> {
+  let summary = "Groups ops into a region that is isolated from above";
+
+  let description = [{
+    Provides a way to group ops into a region that is isolated from above.
+
+    Since the enclosed region is isolated from above, any use of values from
+    outer scopes requires passing that value as an input parameter to this op
+    and using that parameter instead.
+
+    The group_type attribute is a free-form string that can be used to denote
+    any aspect of the group. For example, it could be set to "elementwise" for
+    a group that only contains elementwise ops.
+
```

Comment:
Same comment about the semantics of `group_type`.

Reply:
Removed that attribute.